lib/util/get-typescript-extension-map.js

```diff
@@ -36,46 +55,78 @@ function normalise(typescriptExtensionMap) {
  */
 function get(option) {
     if (
-        option &&
-        option.typescriptExtensionMap &&
-        Array.isArray(option.typescriptExtensionMap)
+        {}.hasOwnProperty.call(tsConfigMapping, option?.typescriptExtensionMap)
     ) {
+        return tsConfigMapping[option.typescriptExtensionMap]
+    }
+
+    if (Array.isArray(option?.typescriptExtensionMap)) {
         return normalise(option.typescriptExtensionMap)
     }
 
     return null
 }
 
+/**
+ * Attempts to get the ExtensionMap from the tsconfig of a given file.
+ *
+ * @param {string} filename - The filename we're getting from
+ * @returns {ExtensionMap} The `typescriptExtensionMap` value, or `null`.
+ */
+function getFromTSConfig(filename) {
+    const tsconfig = getTsconfig(filename, "tsconfig.json", fsCache)
+
+    const jsx = tsconfig?.config?.compilerOptions?.jsx
+    if ({}.hasOwnProperty.call(tsConfigMapping, jsx)) {
+        return tsConfigMapping[jsx]
+    }
+
+    return null
+}
+
 /**
  * Gets "typescriptExtensionMap" setting.
  *
- * 1. This checks `options` property, then returns it if exists.
- * 2. This checks `settings.n` | `settings.node` property, then returns it if exists.
- * 3. This returns `DEFAULT_MAPPING`.
+ * 1. This checks `options.typescriptExtensionMap`, if its an array then it gets returned.
+ * 2. This checks `options.typescriptExtensionMap`, if its a string, convert to the correct mapping.
+ * 3. This checks `settings.n.typescriptExtensionMap`, if its an array then it gets returned.
+ * 4. This checks `settings.node.typescriptExtensionMap`, if its an array then it gets returned.
+ * 5. This checks `settings.n.typescriptExtensionMap`, if its a string, convert to the correct mapping.
+ * 6. This checks `settings.node.typescriptExtensionMap`, if its a string, convert to the correct mapping.
+ * 7. This checks for a `tsconfig.json` `config.compilerOptions.jsx` property, if its a string, convert to the correct mapping.
+ * 8. This returns `PRESERVE_MAPPING`.
  *
- * @param {import('eslint').Rule.RuleContext} context - The rule context.
+ * @param {import("eslint").Rule.RuleContext} context - The rule context.
  * @returns {string[]} A list of extensions.
  */
 module.exports = function getTypescriptExtensionMap(context) {
     return (
-        get(context.options && context.options[0]) ||
-        get(
-            context.settings && (context.settings.n || context.settings.node)
+        get(context.options?.[0]) ||
+        get(context.settings?.n ?? context.settings?.node) ||
+        getFromTSConfig(
+            context.physicalFilename ?? context.getPhysicalFilename()
```

Comment:
This was added in [eslint v7.28.0](https://eslint.org/blog/2021/06/eslint-v7.28.0-released) should I also check `context.getFilename()` for eslint >= 7.0 < 7.28?

Reply:
sounds good. the package supports eslint >= 7.0.0.
https://github.com/eslint-community/eslint-plugin-n/blob/20d2713de7054b823ab29f40925ba782123208c3/package.json#L13-L15